### PR TITLE
Add message button to Profile

### PR DESCRIPTION
**Pull request by elisealix22**
Relates to: https://github.com/xmtp-labs/hq/issues/703

Adds a message button to profiles that sends a `gm` message from the logged in user via XMTP. The test message will be swapped out when we implement the inbox.

<details>
<summary>Screenshots</summary>

| With `messages` FF | Without `messages` FF |
|--|--|
| Screen Shot 2022-10-03 at 3 25 20 PM  |  Screen Shot 2022-10-03 at 3 24 58 PM|

### CODE REVIEW

File: data/feature-flags.ts [hunk 1]
```diff
@@ -1,11 +1,12 @@
 import { aaveMembers } from './aave-members';
 import { lensterMembers } from './lenster-members';
+import { xmtpMembers } from './xmtp-members';
 
 export const featureFlags = [
   {
     key: 'messages',
     name: 'Messages',
-    enabledFor: [...lensterMembers, ...aaveMembers]
+    enabledFor: [...lensterMembers, ...aaveMembers, ...xmtpMembers]
```
Comment (elisealix22) [inline data/feature-flags.ts:9]: Adding your Lens profile is only required for mainnet testing fwiw

File: data/xmtp-members.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+export const xmtpMembers = [
+  '0xe248', // nick-molnar.lens
+  '0x010e04', // elisealix22.lens
+  '0x5cce' // saulmc.lens
```
Comment (neekolas) [inline data/xmtp-members.ts:4]: @bhavya2611 you'll want to add yourself here once you have claimed a Lens Profile

File: src/components/Profile/Details.tsx [hunk 3]
```diff
@@ -91,11 +93,18 @@ const Details: FC<Props> = ({ profile }) => {
...
-              <SuperFollow profile={profile} setFollowing={setFollowing} showText />
+              <div className="flex space-x-2">
+                <SuperFollow profile={profile} setFollowing={setFollowing} showText />
+                {getFeatureEnabled('messages', currentProfile?.id) && <Message profile={profile} />}
+              </div>
             ) : (
-              <Follow profile={profile} setFollowing={setFollowing} showText />
+              <div className="flex space-x-2">
+                <Follow profile={profile} setFollowing={setFollowing} showText />
+                {getFeatureEnabled('messages', currentProfile?.id) && <Message profile={profile} />}
```
Comment (neekolas) [inline src/components/Profile/Details.tsx:106]: Guessing we'll also want to check and make sure you aren't on your own profile. Feels like we should hide in that case

Comment (elisealix22) [reply src/components/Profile/Details.tsx:106]: That check happens above at line 83! https://github.com/xmtp/lenster/pull/1/files#diff-c3be5776a74b374f000912a2470abbff4bf683894146a3b0d212ad80f68a0a5aL81.

We're on the same page -- I did not add the message button on your own profile for now at least 😛

**Review by neekolas — APPROVED** [review]
I'm gonna approve to unblock, but it looks like the button should be purple and have a mail icon
CleanShot 2022-10-03 at 13 28 03@2x

File: src/components/Profile/Details.tsx (hunk 3 above)
Comment (neekolas) [reply src/components/Profile/Details.tsx:106]: Ohh. Missed that. Perfect

**Comment by elisealix22** [comment]
Planning to follow up on styling! I won't close the tracking issue until it's to spec, but hoping to get some scaffolding in place before focusing on colors / icons / spacing 🕵️‍♂️

**Comment by neekolas** [comment]
I think we'll also want to check if the potential recipient is on the XMTP network or not before showing the button.

Maybe the sender too, although we could fall back to a Connect flow if they had not yet enabled.

**Comment by elisealix22** [comment]
Ooooh, that's a great point! Just added a note for that case in the issue: https://github.com/xmtp-labs/hq/issues/703#issuecomment-1266047264. My gut is we'll either want to disable it or show an error message when they tap it if the user isn't on XMTP yet but we'll see what Darick thinks when he's back!

**Pull request merged**